mantidproject/mantid
# Add a toolbar button for rectangle selection on Preview tab

**Description of work.**

This PR adds a toolbar button on the reflectometry gui to initiate custom rectangle selection on the embedded region selector widget. It also:
- does some refactoring of the region selector python widget to allow it to be created without a workspace;
- rename the region selector python tests and add missing tests;
- moves the region selector to the PreviewPresenter and adds an interface so we can mock it
- adds some TODOs related to the next piece of work to add an Edit button;
- does some renaming to remove references to "contour plot" and call them "region selector" or "ROI" instead

Note that this is incomplete as it is part of interim work on an interface that is not exposed to users. Note that the new button does not interact well with the normal sliceviewer toolbar, but that will be removed as this work continues.

Fixes #34170

**To test:**

As well as the following, also do some basic checks on the normal sliceviewer to ensure that it is still working as expected.

- Ensure this workspace is loaded or is in your load path: [INTER45455_inst.zip](https://github.com/mantidproject/mantid/files/7504266/INTER45455_inst.zip)
- Open the ISIS Reflectometry GUI
- Select the Preview tab.
- Type in `INTER45455_inst` and click Load. The instrument viewer plot should display the data.
- Click the rectangle-select button and select a region of interest.
- Whenever the ROI is created or changed, the second plot should be updated with a plot of spectrum vs TOF of the resulting workspace summed across the selected banks. 
- In the second plot, click the rectangle-selection button. It should print the selected spectra (min/max).


*This does not require release notes* because **functionality it is not yet exposed**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
